portfolio-jsm/src/components/Input.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { CSSProperties } from "react";
+import { InputProps } from "@/src/types";
+
+const Input = ({
+  type,
+  placeholder,
+  onChange,
+  width,
+  height,
+  style,
+  inputColor,
+}: InputProps) => {
+  const inputStyle: CSSProperties = {
+    width: width || "100%",
+    height: height || "2.5rem",
+    ...style,
+  };
+
+  return (
+    <input
+      type={type}
+      onClick={
+        type === "Time"
+          ? (e) => {
+              e.currentTarget.showPicker();
+            }
+          : undefined
+      }
```

Comment:
```suggestion
  let onClick;
  
  if (type === "Time") {
    onClick = (e: React.MouseEvent<HTMLInputElement>) => {
      e.currentTarget.showPicker();
    };
  }

  return (
    <input
      type={type}
      onClick={onClick}
      placeholder={placeholder}
      className={`block rounded-[10px] ${inputColor} px-4 py-2`}
      style={inputStyle}
      onChange={onChange}
    />
  );
```

Not a huge fan of this ternary. Consider this

Reply:
I realized I'm not going to use the time picker property so I removed it